### ISSUE TITLE
Removing homeplus pro while HSWidgets is installed == safemode

### ISSUE THREAD

**Issue by bill88t** [DGh0st/HSWidgets#11]
1. Happy bday D_Gh0st!
2. So, after uninstalling homeplus pro I was sent to safe mode due to HSWidgets.
even after removing and reinstalling HSWidgets I am still sent into safe mode.
It appears to be so that it cannot adapt to the difference of the homescreen size.

**Comment by DGh0st** [comment]
> Happy bday D_Gh0st!

Thanks!

Please attach any crash log for any crashes.

> It appears to be so that it cannot adapt to the difference of the homescreen size

The visual size of the homescreen doesn't really matter, but If you meant changing the max amount of icons on the page then I wouldn't be surprised. HSWidgets will gladly allow other tweaks to add more icons, but when it comes to giving up icon spots is when things go hairy.

For instance, if you use a tweak to make the following layout.
```
w = widget space
i = icon space
_ = empty space
```
```
i i i i i
i i i i i
i i i i i
i i i i i
_ _ _ _ _
w w w w w
w w w w w
```
and then you change it to be either of these, its going to get confused (some of these scenarios may actually work currently).
If the number of columns is changed by 1:
```
i i i i i
i i i i i
i i i i i
i i i i i
_ _ _ _ _
w w w w w
w w w w w
```
Should the widget be resized? What if the widget doesn't support resize? Should it be removed instead?

Or changing number of rows by 2:
```
i i i i i
i i i i i
i i i i i
i i i i i
_ _ _ _ _
```
In this case where should the widget that was added go? Should it move itself up and force the icons to the next page?
If we forced the widget up and had the following layout instead:
```
w w w w w
w w w w w
w w w w w
w w w w w
i i i i i
w w w _ _
w w w _ _
```
And then changing number of rows by 2, we get:
```
w w w w w
w w w w w
w w w w w
w w w w w
i i i i i
```
Forcing the widget up wouldn't even fit on the page.
If it were able to fit, lets say with the following scenario:
```
w w w w w
w w w w w
w w w w w
w w w w w
i i i i i
w w w w w
_ _ _ _ _
```
And then again we changed number of rows by 2 and move the widget up, we get this:
```
w w w w w
w w w w w
w w w w w
w w w w w
w w w w w
```
A page full of widgets, which if the user doesn't have a tweak to keep empty homescreen pages then this will remove all the widgets. leaving:
```
i i i i i
_ _ _ _ _
_ _ _ _ _
_ _ _ _ _
_ _ _ _ _
```
I am sure there are lot more edge cases then the ones I mentioned.

**Comment by DGh0st** [comment]
The temporary fix would be to either reinstall HomePlus to remove the widgets that are going to be outside the homescreen grid on uninstall or delete the `/var/mobile/Library/Preferences/com.dgh0st.hswidget.widgetlayouts.plist` then re-add all the widgets.

**Comment by bill88t** [comment]
[crash.txt](https://github.com/DGh0st/HSWidgets/files/5824674/crash.txt)

the situation changed a bit. With the latest version 0.3-4 I was able to uninstall homeplus pro. but after rebooting (yes rebooting) it wouldn't get off of safe mode. It may have to do with velox (and it's entirely different layout) too.. imma troubleshoot it a bit more and see.

**Comment by bill88t** [comment]
[SpringBoard@2021-01-16_5:55_pm.log](https://github.com/DGh0st/HSWidgets/files/5824723/SpringBoard%402021-01-16_5.55_pm.log)
[SpringBoard@2021-01-16_5:56_pm.log](https://github.com/DGh0st/HSWidgets/files/5824724/SpringBoard%402021-01-16_5.56_pm.log)
[SpringBoard@2021-01-16_5:57_pm.log](https://github.com/DGh0st/HSWidgets/files/5824725/SpringBoard%402021-01-16_5.57_pm.log)
[SpringBoard@2021-01-16_6:14_pm.log](https://github.com/DGh0st/HSWidgets/files/5824726/SpringBoard%402021-01-16_6.14_pm.log)
[SpringBoard@2021-01-16_6:16_pm.log](https://github.com/DGh0st/HSWidgets/files/5824727/SpringBoard%402021-01-16_6.16_pm.log)

then after the deletion of `/var/mobile/Library/Preferences/com.dgh0st.hswidget.widgetlayouts.plist`
and the removal of velox I proceeded to go out of safe mode only to be greeted by it again.
Cr4shed really just kept blaming other tweaks. 
Meanwhile, reinstalling HPP HSW and Velox reloaded allowed me do proceed just fine.
Results: HSW without HPP or Velox == safemode

**Comment by DGh0st** [comment]
None of these crashes are due to layout. First one was related to modern weather widget (an off by 1 error that was fixed in 0.3-4). The new 5 crashes are because of IconSupport. IconSupport hasn’t been updated for iOS 13 so there is no reason to have that installed.

**Comment by bill88t** [comment]
the first crash log was done in 0.3-4 and additionally Velox reloaded depends on iconsupport.
I will try removing both tomorrow.

**Comment by bill88t** [comment]
IconSupport was the culprit.
Somehow, while velox and hpp are installed everything's fine. But if they get removed, HSWidgets conflicts with IconSupport.
Now that I removed velox reloaded, IconSupport and Homeplus pro, HSWidgets is working fine. I rebooted to make sure everything's working.

**Comment by DGh0st** [comment]
Closing this issue assuming the first crash log was fixed in 0.3-5 (as I had to re-implement that), please reopen if you see that crash again.
